jdaviz/configs/default/plugins/virtual_observatory/vo_plugin.py

```diff
@@ -0,0 +1,419 @@
+            if self.resource_filter_coverage:
+                try:
+                    # First parse user-provided source as direct coordinates
+                    coord = SkyCoord(
+                        self.source, unit=u.deg, frame=self.coordframe_selected
+                    )
+                except Exception:
+                    try:
+                        # If that didn't work, try parsing it as an object name
+                        coord = SkyCoord.from_name(
+                            self.source, frame=self.coordframe_selected
+                        )
+                    except Exception:
+                        self.resources_loading = False  # Stop loading bar
+                        self.hub.broadcast(
+                            SnackbarMessage(
+                                f"Unable to resolve source coordinates: {self.source}",
+                                sender=self,
+                                color="error",
+                            )
+                        )
+                        raise LookupError(
+                            f"Unable to resolve source coordinates: {self.source}"
+                        )
+                registry_args.append(
+                    registry.Spatial(
+                        (coord, (self.radius_val * u.Unit(self.radius_unit.selected))),
+                        intersect="overlaps",
+                    )
+                )
+            self._full_registry_results = registry.search(*registry_args)
+            self.resources = list(self._full_registry_results.getcolumn("short_name"))
+        except DALFormatError as e:
+            if type(e.cause) is RequestConnectionError:
+                self.hub.broadcast(
+                    SnackbarMessage(
+                        f"Can't connect to VO registry. Check your internet connection: {e}",
+                        sender=self,
+                        color="error",
+                    )
+                )
+            else:
+                raise e
+        except Exception as e:
+            self.hub.broadcast(
+                SnackbarMessage(
+                    f"An error occured querying the VO Registry: {e}",
+                    sender=self,
+                    color="error",
+                )
+            )
+            raise
+        finally:
+            self.resources_loading = False  # Stop loading bar
+
+    def vue_query_resource(self, _=None):
+        """
+        Once a specific VO resource is selected, query it with the user-specified source target.
+        User input for source is first attempted to be parsed as a SkyCoord coordinate. If not,
+        then attempts to parse as a target name.
+        """
+        # Reset Table
+        self.table.items = []
+        self._populate_url_only = False
+        self.table.headers_visible = ["Title", "Instrument", "DateObs"]
+
+        self.results_loading = True  # Start loading spinner
```

Comment:
same as above - could use `with_spinner` decorator.

Reply:
Done in https://github.com/spacetelescope/jdaviz/pull/2872/commits/4993e9284a33081321013bd29e0670c309c99c2f